pyproject.toml

```diff
@@ -75,11 +72,8 @@ geopandas = [
 ]
 
 # Distributed
-modin = [
-    { version = "0.23.1post0", markers = "python_version < \"3.9\"", optional = true },
-    { version = "^0.31.0", markers = "python_version >= \"3.9\"", optional = true }
-]
-ray = { version = "^2.30.0", extras = ["default", "data"], optional = true }
+modin = { version = "^0.31.0", optional = true }
+ray = { version = ">=2.30.0,<2.38.0", extras = ["default", "data"], optional = true }
```

Comment:
Ray has introduced breaking changes to Datasink API. Will open a separate PR for that.